### PR TITLE
121 support deploy metadata

### LINKED ISSUE #121: Support deploy metadata
This feature used to be supported, but was never brought forward in the new 

Want to be able to save the metadata of a widget with the matching name

Reference "deploy_data" here: https://github.com/NEARBuilders/bos-workspace/blob/version/0.0.1-alpha.6/lib/deploy.js#L136

### PR DESCRIPTION
**Pull request by bb-face**
Resolves #121 

This PR adds the function to upload metadata to the social db.
I also added the support for the option `--signerPublicKey` and `--signerPrivateKey` in order to sign the transaction without any user input!

### CODE REVIEW

File: examples/single/bos.config.json
Comment (elliotBraem) [inline examples/single/bos.config.json]: revert this

File: examples/single/widget/home.metadata.json
Comment (elliotBraem) [inline examples/single/widget/home.metadata.json]: Keep this tho

hahah tree version 2

Maybe rename -- single/home is just a BOS implementation of Guest Book

https://github.com/near-examples/guest-book-examples

File: lib/deploy.ts
Comment (elliotBraem) [inline lib/deploy.ts]: social.near in mainnet, v1.social08.testnet in testnet (I think there is a definition that can be exported from server.ts)

File: lib/deploy.ts
Comment (elliotBraem) [inline lib/deploy.ts]: Have this come from network -n

File: lib/deploy.ts
Comment (elliotBraem) [inline lib/deploy.ts]: Remove

File: lib/cli.ts
Comment (elliotBraem) [inline lib/cli.ts]: How about, for consistency:

.argument("[src]", "path to the app source code", ".")

And then account can be derived from config of the app source code, right? But I think this is okay for now, we can adjust when we combine with deploy

**Comment by elliotBraem** [comment]
@race-of-sloths

**Comment by race-of-sloths** [comment]
@bb-face Thank you for your contribution! Your pull request is now a part of the [Race of Sloths](https://race-of-sloths.com)!
New Sloth joined the Race! Welcome!

[<picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://badge.race-of-sloths.com/bb-face?type=bot&theme=dark">
    <source media="(prefers-color-scheme: light)" srcset="https://badge.race-of-sloths.com/bb-face?type=bot&theme=light">
    Shows profile picture for the author of the PR
</picture>](https://race-of-sloths.com/profile/bb-face)





<details><summary>Current status: <i>executed</i></summary>

| Reviewer | Score |
|--------|--------|
| @elliotBraem  | 2 |

**The average score is 2**

@bb-face check out your results on the [Race of Sloths Leaderboard!](https://race-of-sloths.com/leaderboard) and in the [profile](https://race-of-sloths.com/profile/bb-face)

</details><details><summary>What is the Race of Sloths</summary>
<p>
Race of Sloths is a friendly competition where you can participate in challenges and compete with other open-source contributors within your normal workflow

For contributors:

- Tag @race-of-sloths inside your pull requests
- Wait for the maintainer to review and score your pull request
- Check out your position in the [Leaderboard](https://race-of-sloths.com/leaderboard)
- Keep weekly and monthly streaks to reach higher positions
- Boast your contributions with a dynamic picture of your [Profile](https://race-of-sloths.com/profile/bb-face)

For maintainers:
- Score pull requests that participate in the Race of Sloths
- Engage contributors with fair scoring and fast responses so they keep their streaks
- Promote the Race to the point where the Race starts promoting you
- Grow the community of your contributors

Feel free to check [our website](https://race-of-sloths.com) for additional details!
</p>
</details> 

<details>
<summary>Bot commands</summary>

- For contributors
  - **Include a PR:** `@race-of-sloths include` to enter the Race with your PR
- For maintainers:
  - **Assign points:** `@race-of-sloths score [1/2/3/5/8/13]` to award points based on your assessment.
  - **Reject this PR:** `@race-of-sloths exclude` to send this PR back to the drawing board.
  - **Exclude repo:** `@race-of-sloths pause` to stop bot activity in this repo until `@race-of-sloths unpause` command is called

</details>

**Comment by elliotBraem** [comment]
@race-of-sloths score 2

**Comment by race-of-sloths** [comment]
### 🌟 Score recorded!

@elliotBraem, thank you for scoring this pull request in the Race of Sloths!

**Review by elliotBraem — APPROVED** [review]

**Comment by race-of-sloths** [comment]
### ✅ PR is finalized! 
Your contribution is much appreciated with a final score of 2!
You have received 30 (20 base + 10 weekly bonus) Sloth points for this contribution


Another weekly streak completed, well done @bb-face! To keep your weekly streak and get another bonus make pull request next week! Looking forward to see you in race-of-sloths

**Pull request merged**
